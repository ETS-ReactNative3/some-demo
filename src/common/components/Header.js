import React, { useState, useCallback, useRef } from 'react'
import Switch from 'react-switch'
import styled from 'styled-components'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { faBars } from '@fortawesome/free-solid-svg-icons'
import logo from '../../assets/images/airbnb.svg'
import nighIcon from '../../assets/images/night-icon.svg'
import sunnyIcon from '../../assets/images/sunny-icon.svg'

const HeaderContainer = styled.header`
  height: 80px;
  position: fixed;
  width: 100%;
  z-index: 100;
  .header-inner {
    display: flex;
    height: 100%;
    align-items: center;
    padding-left: 24px;
    padding-right: 24px;
    margin-left: auto;
    margin-right: auto
  }
  .header-image {
    height: 32px;
    width: 32px;
  }
  .header-tab {
    flex: 1;
    justify-content: flex-end;
    display: flex
  }
  .theme-icon {
    width: 20px;
    height: 20px;
    margin-top: 4px;
    margin-left: 5px;
  }
  .menu-icon {
    color: ${({ theme }) => theme.highlightText};
    font-size: 28px
  }
  .header-nav-item {
    margin-left: 2rem;
    font-size: 1rem;
    font-weight: 500;
    line-height: 1.4375rem;
    color: ${({ theme }) => theme.highlightText};
    text-decoration: none;
    transition: all 0.3s ease 0s;
  }
  .topnav a.icon {
    display: none;
  }

  @media screen and (max-width: 600px) {
    .topnav a {display: none;}
    .topnav a.icon {
      float: right;
      display: block;
    }
  }
  
  @media screen and (max-width: 600px) {
    .topnav.responsive {
      position: absolute;
      width: 100%;
      left: 0px;
      top: 80px;
      background-color: ${({ theme }) => theme.primaryBackground};
      padding-bottom: 20;
      animation-name: slideUp;
      animation-duration: 0.5s;
      transition-timing-function: ease-in-out;
    }
    .theme-switch {
      margin-right: 10px;
    }
    .theme-switch-active {
      margin-right: 34px;
    }
    .topnav.responsive .icon {
      position: absolute;
      top: -54px;
      right: 24px;
    }
    .topnav.responsive .header-nav-item {
      padding-top: 14px;
      padding-bottom: 14px;
    }
    .topnav.responsive a {
      float: none;
      display: block;
      text-align: left;
    }
  }

  @keyframes slideUp {
    from { left: 60px; }
    to { left: 0px; }
  }
`
const MENU = [
  {
    title: 'Linkedin',
    href: 'https://www.linkedin.com/in/d%C6%B0%C6%A1ng-t%E1%BA%A5n-h%C3%B9ng-633132131/'
  },
  {
    title: 'Github',
    href: 'https://github.com/zrg-team'
  },
  {
    title: 'Source Code',
    href: 'https://github.com/zrg-team/some-demo'
  },
  {
    title: 'Profile',
    href: 'https://zrg-team.github.io/store/'
  }
]
function Header ({ theme, setTheme }) {
  const navRef = useRef(null)
  const switchRef = useRef(null)
  const [selectTheme, setSelectTheme] = useState(theme === 'dark')
  const handleSelectTheme = (checked) => {
    const newTheme = checked ? 'dark' : 'light'
    setTheme(newTheme)
    setSelectTheme(checked)
  }
  const handleToggeMenu = useCallback(() => {
    const element = navRef.current
    if (element.className === 'topnav') {
      element.className += ' responsive'
      switchRef.current.className = 'theme-switch-active'
    } else {
      element.className = 'topnav'
      switchRef.current.className = 'theme-switch'
    }
  }, [navRef])
  return (
    <HeaderContainer>
      <div className='header-inner'>
        <img src={logo} className='header-image' alt='header-image' />
        <div className='header-tab'>
          <div
            ref={switchRef}
            className='theme-switch'
          >
            <Switch
              onChange={handleSelectTheme}
              checked={selectTheme}
              offColor='#555555'
              onColor='#555555'
              // offHandleColor="#0ff"
              // onHandleColor="#08f"
              uncheckedIcon={
                <img src={sunnyIcon} className='theme-icon' alt='sunny-icon' />
              }
              checkedIcon={
                <img src={nighIcon} className='theme-icon' alt='nigh-icon' />
              }
            />
          </div>
          <nav ref={navRef} className='topnav'>
            {
              MENU.map(
                item => (
                  <a
                    className='header-nav-item'
                    key={item.href}
                    href={item.href}
                    title={item.title}
                  >
                    {item.title}
                  </a>
                )
              )
            }
            <a className='icon' onClick={handleToggeMenu}>
              <FontAwesomeIcon icon={faBars} className='menu-icon' />
            </a>
          </nav>
        </div>
      </div>
    </HeaderContainer>
  )
}

export default Header
