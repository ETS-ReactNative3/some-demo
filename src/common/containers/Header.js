import { connect } from 'react-redux'
import Header from '../components/Header'
import commonHandlers from '../handlers/common'

const mapDispatchToProps = (dispatch, props) => ({
  ...commonHandlers(dispatch, props)
})

const mapStateToProps = (state, props) => {
  // TODO: Only map something usefull
  return {
    theme: state.common.theme
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Header)
