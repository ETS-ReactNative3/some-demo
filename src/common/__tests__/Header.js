import React from 'react'
import Header from '../components/Header'
import 'enzyme-adapter-react-16'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

describe('[Component] Header', () => {
  it('[Case] Render with light theme props', () => {
    const wrapper = shallow(
      <Header
        theme='light'
      />
    )

    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('[Case] Render with dark theme props', () => {
    const wrapper = shallow(
      <Header
        theme='dark'
      />
    )

    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})