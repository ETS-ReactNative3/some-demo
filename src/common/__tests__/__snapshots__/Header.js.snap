// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] Header [Case] Render with dark theme props 1`] = `
<styled.header>
  <div
    className="header-inner"
  >
    <img
      alt="header-image"
      className="header-image"
      src="airbnb.svg"
    />
    <div
      className="header-tab"
    >
      <div
        className="theme-switch"
      >
        <ReactSwitch
          activeBoxShadow="0 0 2px 3px #3bf"
          boxShadow={null}
          checked={true}
          checkedIcon={
            <img
              alt="nigh-icon"
              className="theme-icon"
              src="night-icon.svg"
            />
          }
          disabled={false}
          height={28}
          offColor="#555555"
          offHandleColor="#fff"
          onChange={[Function]}
          onColor="#555555"
          onHandleColor="#fff"
          uncheckedIcon={
            <img
              alt="sunny-icon"
              className="theme-icon"
              src="sunny-icon.svg"
            />
          }
          width={56}
        />
      </div>
      <nav
        className="topnav"
      >
        <a
          className="header-nav-item"
          href="https://www.linkedin.com/in/d%C6%B0%C6%A1ng-t%E1%BA%A5n-h%C3%B9ng-633132131/"
          key="https://www.linkedin.com/in/d%C6%B0%C6%A1ng-t%E1%BA%A5n-h%C3%B9ng-633132131/"
          title="Linkedin"
        >
          Linkedin
        </a>
        <a
          className="header-nav-item"
          href="https://github.com/zrg-team"
          key="https://github.com/zrg-team"
          title="Github"
        >
          Github
        </a>
        <a
          className="header-nav-item"
          href="https://github.com/zrg-team/some-demo"
          key="https://github.com/zrg-team/some-demo"
          title="Source Code"
        >
          Source Code
        </a>
        <a
          className="header-nav-item"
          href="https://zrg-team.github.io/store/"
          key="https://zrg-team.github.io/store/"
          title="Profile"
        >
          Profile
        </a>
        <a
          className="icon"
          onClick={[Function]}
        >
          <FontAwesomeIcon
            border={false}
            className="menu-icon"
            fixedWidth={false}
            flip={null}
            icon={
              Object {
                "icon": Array [
                  448,
                  512,
                  Array [],
                  "f0c9",
                  "M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z",
                ],
                "iconName": "bars",
                "prefix": "fas",
              }
            }
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            swapOpacity={false}
            symbol={false}
            title=""
            transform={null}
          />
        </a>
      </nav>
    </div>
  </div>
</styled.header>
`;

exports[`[Component] Header [Case] Render with light theme props 1`] = `
<styled.header>
  <div
    className="header-inner"
  >
    <img
      alt="header-image"
      className="header-image"
      src="airbnb.svg"
    />
    <div
      className="header-tab"
    >
      <div
        className="theme-switch"
      >
        <ReactSwitch
          activeBoxShadow="0 0 2px 3px #3bf"
          boxShadow={null}
          checked={false}
          checkedIcon={
            <img
              alt="nigh-icon"
              className="theme-icon"
              src="night-icon.svg"
            />
          }
          disabled={false}
          height={28}
          offColor="#555555"
          offHandleColor="#fff"
          onChange={[Function]}
          onColor="#555555"
          onHandleColor="#fff"
          uncheckedIcon={
            <img
              alt="sunny-icon"
              className="theme-icon"
              src="sunny-icon.svg"
            />
          }
          width={56}
        />
      </div>
      <nav
        className="topnav"
      >
        <a
          className="header-nav-item"
          href="https://www.linkedin.com/in/d%C6%B0%C6%A1ng-t%E1%BA%A5n-h%C3%B9ng-633132131/"
          key="https://www.linkedin.com/in/d%C6%B0%C6%A1ng-t%E1%BA%A5n-h%C3%B9ng-633132131/"
          title="Linkedin"
        >
          Linkedin
        </a>
        <a
          className="header-nav-item"
          href="https://github.com/zrg-team"
          key="https://github.com/zrg-team"
          title="Github"
        >
          Github
        </a>
        <a
          className="header-nav-item"
          href="https://github.com/zrg-team/some-demo"
          key="https://github.com/zrg-team/some-demo"
          title="Source Code"
        >
          Source Code
        </a>
        <a
          className="header-nav-item"
          href="https://zrg-team.github.io/store/"
          key="https://zrg-team.github.io/store/"
          title="Profile"
        >
          Profile
        </a>
        <a
          className="icon"
          onClick={[Function]}
        >
          <FontAwesomeIcon
            border={false}
            className="menu-icon"
            fixedWidth={false}
            flip={null}
            icon={
              Object {
                "icon": Array [
                  448,
                  512,
                  Array [],
                  "f0c9",
                  "M16 132h416c8.837 0 16-7.163 16-16V76c0-8.837-7.163-16-16-16H16C7.163 60 0 67.163 0 76v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16zm0 160h416c8.837 0 16-7.163 16-16v-40c0-8.837-7.163-16-16-16H16c-8.837 0-16 7.163-16 16v40c0 8.837 7.163 16 16 16z",
                ],
                "iconName": "bars",
                "prefix": "fas",
              }
            }
            inverse={false}
            listItem={false}
            mask={null}
            pull={null}
            pulse={false}
            rotation={null}
            size={null}
            spin={false}
            swapOpacity={false}
            symbol={false}
            title=""
            transform={null}
          />
        </a>
      </nav>
    </div>
  </div>
</styled.header>
`;
